pypdf/_reader.py

```diff
@@ -649,7 +649,23 @@ def _find_eof_marker(self, stream: StreamType) -> None:
         """
         HEADER_SIZE = 8  # to parse whole file, Header is e.g. '%PDF-1.6'
         line = b""
+        first = True
         while line[:5] != b"%%EOF":
+            if line != b"" and first:
+                if any(
+                    line.strip().endswith(tr) for tr in (b"%%EO", b"%%E", b"%%", b"%")
```

Comment:
Shouldn't we cover `%%EO` as well?

Reply:
Unless i'm wrong, I cover all cases (see test)